scipy/stats/_stats.pyx

```diff
@@ -300,3 +301,165 @@ def _weightedrankedtau(ordered[:] x, ordered[:] y, intp_t[:] rank, weigher, bool
     tau = ((tot - (v + u - t)) - 2. * exchanges_weight[0]
            ) / np.sqrt(tot - u) / np.sqrt(tot - v)
     return min(1., max(-1., tau))
+
+
+# FROM MGCPY: https://github.com/neurodata/mgcpy
+
+# Distance transforms used for MGC and Dcorr
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cpdef _dense_rank_data(np.ndarray[np.float_t, ndim=1] x):
```

Comment:
here and in a few other places `cpdef` functions are defined, but it seems the functions are only used in Cython--perhaps there is some room for simplification by only using `cdef` when calls from Python aren't used / envisioned?

Reply:
https://github.com/scipy/scipy/pull/10524/commits/a9e9b1f5a7673163a2c1d046c72c2c692a212212